### PR TITLE
1051 - add initial git commit signing instructions and example images

### LINKED ISSUE #1051: Add instructions to configure git commit signing to the guidebook
Potential location is https://guidebook.civicactions.com/en/latest/common-practices-tools/security/encryption/

### PR DESCRIPTION
**Pull request by sfmcgee**
----
:books: Documentation preview :books:: https://civicactions-handbook--1101.org.readthedocs.build/en/1101/

### CODE REVIEW

File: practice-areas/engineering/git.md [hunk 1]
```diff
@@ -147,6 +147,151 @@ Here is a recommended Git configuration block that it is recommended to add to y
...
   ff = only
+[commit]
+  gpgsign = true
+```
+
+## Commit Signing
+
+Commit signing allows for identification and non-repudiation of individual commits to repositories in both Github and Gitlab. It improves security of our repositories by ensuring we know who made changes. The following instructions describe how to create a new GPG keypair, associate it with your Github and Gitlab accounts, and enable commit signing by default with git.
+
+### Creating a new GPG Key
```
Comment (openprivacy) [inline practice-areas/engineering/git.md]: This section is excellent and IMO should get its own page ("Setting up GnuPG") for GPG in general, and here keep just the lines that are specific to the `.gitconfig` setup and other Git-related commands. I would link to the general GnuPG page from https://guidebook.civicactions.com/en/latest/common-practices-tools/security/encryption/#the-gnu-privacy-guard-gpg

File: practice-areas/engineering/git.md [hunk 2]
```diff
@@ -147,6 +147,151 @@ Here is a recommended Git configuration block that it is recommended to add to y
...
+
+```shell
+gpg --list-secret-keys --keyid-format=long
+
+gpg: checking the trustdb
+gpg: marginals needed: 3  completes needed: 1  trust model: pgp
+gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
+/home/username/.gnupg/pubring.kbx
+-------------------------------
+sec   rsa4096/FF94245366AB39B2 2022-09-13 [SC]
```
Comment (openprivacy) [inline practice-areas/engineering/git.md]: Even though it is just a fingerprint, consider randomizing the values here (and maybe blur it out in the assets above). You might want to blur our the "mcgee" part of your name, too -- this is a public page. Up to you (unless there is Guidebook guidance on this.)

**Review by ContentBrewmaster — APPROVED** [review]

File: practice-areas/engineering/git.md (hunk 2 above)
Comment (ContentBrewmaster) [reply practice-areas/engineering/git.md]: @sfmcgee did you want to consider any of @openprivacy's comments before we merge this?

Comment (sfmcgee) [reply practice-areas/engineering/git.md]: @ContentBrewmaster Yes, actually I would, thank you. I just haven't had a chance to work on those updates yet.

File: common-practices-tools/security/gnupg.md
Comment (dmundra) [inline common-practices-tools/security/gnupg.md:1]: @sfmcgee you want to add this file to the navigation in the 'mkdocs.yml' file.

Comment (sfmcgee) [reply common-practices-tools/security/gnupg.md:1]: Thanks @dmundra , updated.

File: practice-areas/engineering/git.md (hunk 1 above)
Comment (sfmcgee) [reply practice-areas/engineering/git.md]: Ok, thanks @openprivacy! I split it out into a new gnupg.md page. Let me know if it works.

File: practice-areas/engineering/git.md (hunk 2 above)
Comment (sfmcgee) [reply practice-areas/engineering/git.md]: @openprivacy - thanks for the suggestion, I updated the fingerprints and the images.

**Review by openprivacy — APPROVED** [review]
@sfmcgee This looks great! Thank you for making these changes.

**Pull request merged**